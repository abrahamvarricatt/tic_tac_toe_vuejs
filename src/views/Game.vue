<template>
  <div>
    <!-- Title -->
    <div class="row">
      <div class="col text-center">
        <h1 class="display-4 title">Game Page</h1>
        <br />
      </div>
    </div>
    <!-- Game Board -->
    <div class="row">
      <div class="col">
        <table>
          <tbody>
            <tr>
              <td>
                <gamebutton
                  :x="0"
                  :y="0"
                  :display="game_world[0][0]"
                  @game-click="processClick"
                />
              </td>
              <td>
                <gamebutton
                  :x="0"
                  :y="1"
                  :display="game_world[0][1]"
                  @game-click="processClick"
                />
              </td>
              <td>
                <gamebutton
                  :x="0"
                  :y="2"
                  :display="game_world[0][2]"
                  @game-click="processClick"
                />
              </td>
            </tr>
            <tr>
              <td>
                <gamebutton
                  :x="1"
                  :y="0"
                  :display="game_world[1][0]"
                  @game-click="processClick"
                />
              </td>
              <td>
                <gamebutton
                  :x="1"
                  :y="1"
                  :display="game_world[1][1]"
                  @game-click="processClick"
                />
              </td>
              <td>
                <gamebutton
                  :x="1"
                  :y="2"
                  :display="game_world[1][2]"
                  @game-click="processClick"
                />
              </td>
            </tr>
            <tr>
              <td>
                <gamebutton
                  :x="2"
                  :y="0"
                  :display="game_world[2][0]"
                  @game-click="processClick"
                />
              </td>
              <td>
                <gamebutton
                  :x="2"
                  :y="1"
                  :display="game_world[2][1]"
                  @game-click="processClick"
                />
              </td>
              <td>
                <gamebutton
                  :x="2"
                  :y="2"
                  :display="game_world[2][2]"
                  @game-click="processClick"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</template>

<script>
import gamebutton from '../components/GameButton'
import { mapActions } from 'vuex'

export default {
  components: {
    gamebutton
  },
  computed: {
    game_world: function() {
      return this.$store.getters.gameworld
    }
  },
  methods: {
    ...mapActions(['play_move']),
    processClick: function({ x, y }) {
      console.log('button was clicked at these co-ordinates = ' + x + ',' + y)
      this.play_move({ x: x, y: y, display: 'X' })
    }
  }
}
</script>

<style lang="scss" scoped>
table {
  @extend .mx-auto;
  border: 3px;
  border-style: solid;
}
table tr {
  border: 3px;
  border-style: solid;
}
table tr td {
  border: 3px;
  border-style: solid;
}

button {
  margin: 10px;
  width: 70px;
  height: 70px;
}
</style>
