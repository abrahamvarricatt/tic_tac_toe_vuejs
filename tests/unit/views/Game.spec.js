import { shallowMount } from '@vue/test-utils'
import Game from '@/views/Game'

describe('Game.vue', () => {
  it('renders correctly', () => {
    let wrapper = shallowMount(Game)
    expect(wrapper.element).toMatchSnapshot()
  })
})
