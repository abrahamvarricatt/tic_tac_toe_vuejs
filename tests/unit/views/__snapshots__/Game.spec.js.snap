// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game.vue renders correctly 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col text-center"
    >
      <h1
        class="display-4 title"
      >
        Game Page
      </h1>
       
      <br />
    </div>
  </div>
   
  <div
    class="row"
  >
    <div
      class="col"
    >
      <table>
        <tbody>
          <tr>
            <td>
              <gamebutton-stub
                display="O"
                x="0"
                y="0"
              />
            </td>
             
            <td>
              <gamebutton-stub
                display="X"
                x="0"
                y="1"
              />
            </td>
             
            <td>
              <gamebutton-stub
                display=""
                x="0"
                y="2"
              />
            </td>
          </tr>
           
          <tr>
            <td>
              <gamebutton-stub
                display=""
                x="1"
                y="0"
              />
            </td>
             
            <td>
              <gamebutton-stub
                display=""
                x="1"
                y="1"
              />
            </td>
             
            <td>
              <gamebutton-stub
                display="X"
                x="1"
                y="2"
              />
            </td>
          </tr>
           
          <tr>
            <td>
              <gamebutton-stub
                display="X"
                x="2"
                y="0"
              />
            </td>
             
            <td>
              <gamebutton-stub
                display="X"
                x="2"
                y="1"
              />
            </td>
             
            <td>
              <gamebutton-stub
                display="X"
                x="2"
                y="2"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
